scipy/optimize/tests/test_trustregion_krylov.py

```diff
@@ -0,0 +1,103 @@
+"""
+Unit tests for Krylov space trust-region subproblem solver.
+
+To run it in its simplest form::
+  nosetests test_optimize.py
+
+"""
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.optimize._trlib import (TRLIBQuadraticSubproblem)
+from numpy.testing import (TestCase, assert_, assert_array_equal,
+                           assert_almost_equal,
+                           assert_equal, assert_array_almost_equal,
+                           assert_array_less, run_module_suite)
+
+KrylovQP = TRLIBQuadraticSubproblem(tol_rel_i=1e-8, tol_rel_b=1e-6)
+
+class TestKrylovQuadraticSubproblem(TestCase):
+
+    def test_for_the_easy_case(self):
```

Comment:
Thanks for the suggestion. I have added the first and third test, but not the second and the third for a gradient that is further away from zero.

The reason for that is a limitation of the method: As the problem is solved within the Krylov subspace generated by the hessian and the gradient, this subspace is trivial for a zero gradient and thus the method can't be applied to the case where the gradient is zero as it can only find the zero vector. Also the Krylov subspace buildup is ill-conditioned for tiny initial gradient.
These are inherent limitations of every Krylov subspace based method (most methods that make use of matrix vector products with the hessian) and not the implementation.

From an optimization point of view this is not a real drawback since tiny gradient means convergence already and any senseful optimization algorithm should check for this and stop before passing such a subproblem to the solver.

As a side node: The C trlib package has a lot more test cases.